aws-athena/views/census-vw_acs5_stat.sql

```diff
@@ -1,7 +1,7 @@
 -- View to convert raw ACS5 variables into useable statistics
 WITH distinct_years AS (
     SELECT DISTINCT year
-    FROM spatial.parcel
+    FROM {{ ref('parcel') }}

```

Comment:
Somewhat annoyingly, we can't specify schemas when we select from dbt `ref`s; dbt requires that [model names be unique](https://docs.getdbt.com/faqs/models/unique-model-names), with an [alias system](https://docs.getdbt.com/reference/resource-configs/alias) for models that share the same name in different schemas, and as a result the `ref()` interface only accepts a unique identifier and can't be qualified with a schema. I don't love this interface -- it's helpful to know where a model lives in the schema structure when reading query code, and I don't like that we lose that information by switching to `ref`s.

There is a [two-argument variant of `ref()`](https://docs.getdbt.com/reference/dbt-jinja-functions/ref#two-argument-variant), but the second argument specifies the dbt "project" that the model comes from, not the schema (or "database" in dbt lingo). We could refactor our models so that they live in different projects, but then we would end up duplicating our project configuration files, so it doesn't really feel like the right abstraction.

If we find the lack of schemas in these references to be unacceptable, another option would be to more widely leverage aliases and prefix every model identifier with the schema it lives under, e.g.:

```suggestion
    FROM {{ ref('schema_parcel') }}
```

This would be similar to the way that the views are organized in the `aws-athena/views/` directory, and would allow us to reference the schema for a model without having to change the name of the generated view/table in Athena. We do this already in the code below for a few models whose table names collide with a table in a different schema. However, this approach also comes with downsides, namely:

1. The prefixing convention may not be any clearer than the unique-tablename convention; and
2. SQLFluff [considers hyphens to be unparseable](https://github.com/sqlfluff/sqlfluff/issues/3501), so we would have to turn it off and switch to sqlfmt for these view definitions

I'm open to different options here, curious what you think!

Reply:
@dfsnow Based on [the section of the docs where dbt explains how they organize projects](https://docs.getdbt.com/guides/best-practices/how-we-structure/2-staging#staging-files-and-folders), it seems like a common naming convention that could support schemas is `[source]__[entity].sql`. This isn't _exactly_ the same use case as ours; dbt uses this convention to indicate the "source" (i.e. origin database) of a "staging" model (i.e. initial transformation), whereas we would be using it to indicate the schemas of all our models, regardless of whether they're intended for initial or final transformations. But I think that their guidance for how to organize projects is overly complicated for our use case, and I'm more interested in the way that `[source]__[entity].sql` seems to be an existing standard for namespacing entities in dbt.

Following this pattern, this ref would look like:

```suggestion
    FROM {{ ref('spatial__parcel') }}
```

After seeing it used in their sample code, I am somewhat reassured that it's not that confusing of a pattern, and it's largely similar to how `aws-athena/views/` is organized anyway. The downside would be some amount of duplication between our subdirectory structure and our file prefixing pattern, but that feels like a negligible concern to me. What do you think?

Reply:
Let's go with `[source]__[entity].sql` then, since it's a common pattern, would be consistent across all tables, and maintains the semantic info added by `[location].[table]`.

TBD on whether we should keep the symlink system or eventually move all the model definitions into the dbt subdirectory.